scripts/fix-user.sh

```diff
@@ -0,0 +1,55 @@
+CUID=${1:-$(id -u)}
+CGID=${2:-$(id -g)}
+CUSER=${3:-www-data}
+CGROUP=${4:-www-data}
+# COMMAND_PREFIX may be used to run inside or outside container
+# it can be docker-compose exec -T {containername} or empty string to run locally
+COMMAND_PREFIX=${5:-'docker-compose exec -T php '}
+echo Setting User ${CUSER} = ${CUID} and Group ${CGROUP} = ${CGID}
+
+if [[ $(${COMMAND_PREFIX} grep -c ${CUSER}:x:${CUID}:${CGID} /etc/passwd) == 1 ]]; then
+    echo ${CUSER} already is ${CUID}:${CGID}. No needs to modify users inside container.
+else
+    set -x
+    # Install shadow if it is not exeists as we need usermod and groupmod
+    # For now it is done for Alpine's APK, and not sure is it needed for other OS
+    if [[ (($(${COMMAND_PREFIX} which groupmod) == "") && ($(${COMMAND_PREFIX} which apk) != "")) ]]; then
+        ${COMMAND_PREFIX} apk add --no-cache shadow
+    fi;
+
+    # groupname for group with GID=CGID
+    EXISTENT_GROUP_NAME=$(${COMMAND_PREFIX} getent group ${CGID} | cut -d: -f1)
+    # username for user with UID=CUID
+    EXISTENT_USER_NAME=$(${COMMAND_PREFIX} getent passwd ${CUID} | cut -d: -f1)
+    # Actual UID and GID for CUSER
+    EXISTENT_UID=$(${COMMAND_PREFIX} getent passwd ${CUSER} | cut -d: -f3)
+    EXISTENT_GID=$(${COMMAND_PREFIX} getent passwd ${CGROUP} | cut -d: -f3)
+
+    # Moving existent GID to new one, if any exists
+    if [[ (${EXISTENT_GROUP_NAME} != ${CGROUP}) && (${EXISTENT_GROUP_NAME} != '') ]]; then
+        # Getting next free GID bigger, then 1000
+        NEW_GID=$(${COMMAND_PREFIX} awk -F: '{gid[$3]=1}END{for(x=1000; x<=10000; x++) {if(gid[x] != ""){}else{print x; exit;}}}' /etc/group)
+        ${COMMAND_PREFIX} groupmod ${EXISTENT_GROUP_NAME} -g ${NEW_GID}
+        ${COMMAND_PREFIX} find / -group ${CGID} -exec chgrp -h ${NEW_GID} {} \;
```

Comment:
not sure it's good idea to update all files because target user could be root (0 - needs separate check)  and you will change system critical files a-la `/dev` `/proc` `/sys`!!!

Reply:
updated